src/rdf4cpp/rdf/namespaces/util/NamespaceInstance.hpp

```diff
@@ -12,18 +12,20 @@ template<typename NamespaceClass>
 struct NamespaceInstance {
     static NamespaceClass &instance(storage::node::NodeStorage &node_storage = storage::node::NodeStorage::default_instance()) {
         static storage::util::tsl::sparse_map<uint16_t, NamespaceClass> instances;
```

Comment:
Given that NodeStorage is thread-safe now, it would probably make sense to make Namespaces threadsafe as well.